airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test-integration/java/io/airbyte/integrations/source/mysql_strict_encrypt/AbstractMySqlSslCertificateStrictEncryptSourceAcceptanceTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mysql_strict_encrypt;
+
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.Database;
+import io.airbyte.db.MySqlUtils;
+import io.airbyte.db.factory.DSLContextFactory;
+import io.airbyte.db.factory.DatabaseDriver;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.integrations.source.mysql.MySqlSource.ReplicationMethod;
+import io.airbyte.integrations.standardtest.source.TestDestinationEnv;
+import org.jooq.DSLContext;
+import org.jooq.SQLDialect;
+import org.testcontainers.containers.MySQLContainer;
+
+public abstract class AbstractMySqlSslCertificateStrictEncryptSourceAcceptanceTest extends MySqlStrictEncryptSourceAcceptanceTest {
+
+  protected static MySqlUtils.Certificate certs;
+  protected static final String PASSWORD = "Passw0rd";
+
+  @Override
+  protected void setupEnvironment(final TestDestinationEnv environment) throws Exception {
+
+    container = new MySQLContainer<>("mysql:8.0");
```

Comment:
where do we create users?
We can enforce client certificate on the test user by creating using `REQUIRE X509`.

Reply:
the user is created in the `MySQLContainer` class. For these tests, I use certificates that are already in the test docker container. The certificates are in the container in the following path - `/var/lib/mysql/`. There are - root certificate, server certificates and client certificates. Then, I add them to the `mysql` configuration - to the` /etc/my.cnf` file. I do this in the `io.airbyte.db.MySqlUtils` class in the getCertificate method. Without configuring the `my.cnf` file - SSL will not work. As for client certificates, there are several ways to configure the server to work with them. If you add the paths to the client certificates to the main /etc/my.cnf file, then any user can connect with these client certificates. Another way is to create a `my.cnf` file in the user's directory and add paths to client certificates to it. In this case, only one specific user will be able to connect with this and other client certificates.

Reply:
Creating a user on mysql with `REQUIRE X509` will actually create a user for which login with any _valid_ client certificate should work. 
In real life you would put a more granular `REQUIRE` making sure that the certificate userX is sending actually has userX on the subject. Something like:
`CREATE USER 'jeffrey'@'localhost'
  REQUIRE SUBJECT '/C=SE/ST=Stockholm/L=Stockholm/
    O=MySQL demo client certificate/
    CN=client/emailAddress=client@example.com';`

This may be a bigger change that can go to another task, but ideally we would want to test here with a user that can only login with a cert

Reply:
I understand that such a case will look good here. It is better to make it a separate task, so as not to waste time on it now, and give it to end users sooner. My local mysql is currently configured to run the testssl user with certificates only - and it works fine for it. But it is not yet implemented in our tests. I created a ticket to fix this - https://github.com/airbytehq/airbyte/issues/15639.